### DISCUSSION TITLE
How to create a snapshot with docker and cli?

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#16988] [Archive]
I’ve set up directus with docker on my local machine. (On previous projects I’ve used the CLI to set up directus, but according to [this comment](https://github.com/directus/docs/issues/247#issuecomment-1366484123), I’ll use docker compose in the future instead)

I’m trying to figure out how to integrate an useful way to create snapshots. According to the docs I have to use `npx directus schema snapshot ./snapshot.yaml`, which only worked with the CLI-Setup. Now with docker compose I have to run `docker exec -it $CONTAINER_ID sh` and then paste the command. But then the snapshot is created inside the directus container and not on my local machine. Of course, I could create another volume for `/snapshots`, but I always have to do it from within the container …

Is there a more comfortable way to do this?

I really liked the previous approach with adding a custom script to `package.json`, like described [here in the docs](https://docs.directus.io/reference/cli.html#snapshot-the-data-model)

**Comment by br41nslug** [comment]
I tend to add a snapshots volume and handle it pretty similar to the npm approach from there:
- add the volume in your docker-compose.yml
   ``` - './snapshots:/directus/snapshots'```
- run `docker compose exec directus npx directus schema snapshot /directus/snapshots/test.yml`
- now the new snapshot should be available in your locally mounted volume.

You can add the command above to a package.json too if you have one in the same folder as the `docker-compose` file. From that folder you should be able to run the commands you're used to but they need to be prefixed with `docker compose exec directus <command here>` in order to run in the docker instead of locally. (presuming your Directus service is called directus)

**Reply by ghost** [reply]
Thanks a lot! That’s a great answer!

**Reply by ulvidamirli** [reply]
@br41nslug Thank you for your reply. But when I run this on production, I get this error in return: 
Error: EACCES: permission denied, open '/directus/snapshots/test.yml'

Do you know how can I solve this issue? In development, no issues

**Reply by br41nslug** [reply]
@ulvidamirli  Have you checked the permissions on the snapshot mount? Presumably created as root

**Reply by ulvidamirli** [reply]
@br41nslug I can't find how I can check it. Could you please help me on that? I use docker compose on non-root user. My OS on production is Ubuntu

**Reply by br41nslug** [reply]
There should be plenty of guides available online to help you figure out linux file permissions (https://kb.iu.edu/d/abdb for instance). But as you mention it is a production server i'd recommend leaving that up to the system admin for that server.

**Reply by ulvidamirli** [reply]
@br41nslug Thank you for the link and the guidance. I solved it by removing and recreating the 'snapshots' folder with the non-root user. And directus binded it to the 'node' user inside container (as simple as this).  I wrote this here to help if somebody face the same issue.

**Comment by hamilton-lima** [comment]
Given that your docker-compose file have this configuration
```
  directus:
    container_name: directus
    image: directus/directus:10.8.0
...
    volumes:
      - directus-volume:/directus/snapshots
```

You can run run this to generate the snapshot 
`docker exec -it directus npx directus schema snapshot /directus/snapshots/current.yml`

And this to copy to your local machine 
`docker cp directus:/directus/snapshots/current.yml .`
